[BUG] Delta Lake deletion vector tests failing on Spark != 3.5 due to version incompatibilities

**Describe the bug**
Build: 
rapids_it-matrix-pre_release-github/201
rapids_it-arm64-pre_release/282
rapids_it-AQE-pre_release-github/505

New Delta Lake deletion vector tests added in commit 9c95eaaff25b3b24f8bee64727e1fd29785c2add are failing across multiple spark versions with version-specific errors:

- Spark 3.2.4: DELETE command returns empty list causing IndexError when accessing result[0][0]
- Spark 3.3.4: (a) Partition split assertion fails (2 vs 1 expected), (b) delta.enableDeletionVectors config not recognized, (c) REORG command syntax not supported
- Spark 3.4.4: IllegalArgumentException when reading __delta_internal_is_row_deleted metadata column due to ColumnarToRow execution plan issue
- Spark 4.0.0: `test_delta_filter_out_metadata_col` - deletion vector metadata column `__delta_internal_is_row_deleted` not found in CPU query plan
- Spark 4.0.1: `test_delta_scan_split_with_DV_enabled_after_DVs_materialized` - NoSuchMethodError for ParquetToSparkSchemaConverter during REORG PURGE

Tests affected:
- test_delta_scan_split_with_DV_enabled_with_DVs
- test_delta_scan_split_with_DV_disabled_with_DVs
- test_delta_scan_split_with_DV_enabled_after_DVs_materialized
- test_delta_filter_out_metadata_col

spark 3xx Error logs:
```
[Spark 3.2.4] IndexError: list index out of range at delta_lake_test.py:169
[Spark 3.3.4] AssertionError: Expected 1 partitions, got 2 at delta_lake_test.py:144
[Spark 3.3.4] AnalysisException: Unknown configuration: delta.enableDeletionVectors
[Spark 3.3.4] ParseException: Syntax error 'REORG' at delta_lake_test.py:195
[Spark 3.4.4] IllegalArgumentException: Part of plan not columnar ColumnarToRowExec
```

spark 40x Error logs:
```
Spark 4.0.0: assert '__delta_internal_is_row_deleted' in query plan failed
Spark 4.0.1: java.lang.NoSuchMethodError: 'void org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(boolean, boolean, boolean, boolean, boolean)' at ReorgTableHelper.filterParquetFiles
```


**Environment details**
Spark versions: 3.2.x, 3.3.x, 3.4.x, 4.0.x
Test framework: pytest with GPU integration tests
Delta Lake version: varies by Spark version

--- a/integration_tests/src/main/python/delta_lake_test.py
+++ b/integration_tests/src/main/python/delta_lake_test.py
@@ -277,13 +277,16 @@ def test_delta_name_column_mapping_no_field_ids(spark_tmp_path, enable_deletion_
 def test_delta_filter_out_metadata_col(spark_tmp_path):
     data_path = spark_tmp_path + "/DELTA_DATA"
 
+    col_a_gen = IntegerGen(min_val=0, max_val=100, nullable=False, special_cases=[])
+    col_b_gen = IntegerGen(min_val=0, max_val=1, nullable=False, special_cases=[0, 1])
+
     def create_delta(spark):
-        two_col_df(spark, int_gen, int_gen).coalesce(1).write.format("delta") \
+        two_col_df(spark, col_a_gen, col_b_gen, length=4000).coalesce(1).write.format("delta") \
             .option("delta.enableDeletionVectors", "true") \
             .partitionBy("a").save(data_path)
 
         count = spark.sql(f"DELETE FROM delta.`{data_path}` WHERE b = 0").collect()[0][0]
-        assert(count > 0)
+        assert count > 100, "Expected enough rows to be deleted to create deletion vectors"
 
     def read_table(spark):
         df = spark.sql(f"SELECT * FROM delta.`{data_path}`")

